### PR TITLE
Rework all routing to use the new artificial key and partition number IOW-562

### PR DESCRIPTION
**Pull request by kkehl-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Rework all routing and proprocessing takes to use the new artificial key and partition number
-----------
Parse the UUID from the S3 object key and insert it into the json_data table.
https://internal.cida.usgs.gov/jira/browse/IOW-562

Description
-----------

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

**Comment by kkehl-usgs** [comment]
@ssoper-usgs @ayan-usgs 

This is okay to review now.

File: src/etl/rds.py [hunk 1]
```diff
@@ -203,6 +216,13 @@ def validate_json(self, variable_name, actual):
         except Exception:
             raise ValidationException("Must be JSON", variable_name, "expected valid JSON", actual)
 
+    def validate_uuid(self, actual):
+        try:
+            uuid_obj = UUID(actual, version=4)
+        except ValueError:
+            raise ValidationException("Must be UUID4", "uuid", "expected valid UUID", actual)
+
+
```
Comment (ayan-usgs) [inline src/etl/rds.py]: Very cool.

File: src/etl/rds.py [hunk 2]
```diff
@@ -116,9 +121,17 @@ def persist_data(self, datum):
                 convert_total_seconds_to_datetime(datum.start_time),
                 convert_total_seconds_to_datetime(datum.response_time), int(datum.response_code),
                 datum.url, api, datum.script_name, int(datum.script_pid),
-                datum.parameters, datum.content
+                datum.parameters, datum.content, datum.uuid
             )
         )
+
+        # TBD IOW-561 will import S3 objects directly into RDS.  All of them? Or only the big ones?
```
Comment (ayan-usgs) [inline src/etl/rds.py:129]: All of them.

**Review by ayan-usgs — APPROVED** [review]

File: src/etl/rds.py [hunk 3]
```diff
@@ -103,11 +105,14 @@ def persist_data(self, datum):
...
+        # ensure that uuid is in uuid4 format
+        self.validate_uuid(datum.uuid)
+
         insert_json_data = """
             INSERT INTO capture.json_data 
             (start_time, response_time, response_code, url, api,
              script_name, script_pid,
-             parameters, json_content)
+             parameters, json_content, uuid)
             VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)
```
Comment (ayan-usgs) [inline src/etl/rds.py]: With `uuid` added, another `%s` should be added as well.

**Review by ayan-usgs — CHANGES_REQUESTED** [review]

File: src/etl/rds.py (hunk 3 above)
Comment (kkehl-usgs) [reply src/etl/rds.py]: Fixed and working on test now ...

File: src/etl/event_processor.py [hunk 4]
```diff
@@ -43,6 +43,8 @@ def __init__(self, record, region):
         self._data = json.loads(self.s3.get_file(self._bucket_name, self._object_key))
         self.metadata = self._data['metadata']
         self.content = self._data['content']
+        proto_uuid = self._object_key.replace(".json", "")
+        self.uuid = proto_uuid[-36:]
```
Comment (ayan-usgs) [inline src/etl/event_processor.py:47]: This could all be done in one line, but that's matter of personal preference. :man_shrugging:

**Review by ayan-usgs — APPROVED** [review]
Looks good to go.

**Pull request merged**